sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/block/CompactBlockDbStore.kt

```diff
@@ -5,58 +5,62 @@ import androidx.room.Room
 import kotlinx.coroutines.withContext
+import kotlin.math.max
 
 /**
  * An implementation of CompactBlockStore that persists information to a database in the given
  * path. This represents the "cache db" or local cache of compact blocks waiting to be scanned.
- *
- * @param appContext the application context. This is used for creating the database.
- * @property dbPath the absolute path to the database.
  */
-class CompactBlockDbStore(
-    appContext: Context,
-    val dbPath: String
-) : CompactBlockStore {
-
-    private val cacheDao: CompactBlockDao
+class CompactBlockDbStore private constructor(
+    private val cacheDao: CompactBlockDao,
     private val cacheDb: CompactBlockDb
+) : CompactBlockStore {
 
-    init {
-        cacheDb = createCompactBlockCacheDb(appContext)
-        cacheDao = cacheDb.complactBlockDao()
-    }
-
-    private fun createCompactBlockCacheDb(appContext: Context): CompactBlockDb {
-        return Room.databaseBuilder(appContext, CompactBlockDb::class.java, dbPath)
-            .setJournalMode(RoomDatabase.JournalMode.TRUNCATE)
-            // this is a simple cache of blocks. destroying the db should be benign
-            .fallbackToDestructiveMigration()
-            .build()
-    }
-
-    override suspend fun getLatestHeight(): Int = withContext(SdkDispatchers.IO) {
-        Math.max(0, cacheDao.latestBlockHeight())
-    }
+    override suspend fun getLatestHeight(): Int = max(0, cacheDao.latestBlockHeight())
 
-    override suspend fun findCompactBlock(height: Int): CompactFormats.CompactBlock? {
-        return cacheDao.findCompactBlock(height)?.let { CompactFormats.CompactBlock.parseFrom(it) }
-    }
+    override suspend fun findCompactBlock(height: Int): CompactFormats.CompactBlock? =
+        cacheDao.findCompactBlock(height)?.let { CompactFormats.CompactBlock.parseFrom(it) }
 
-    override suspend fun write(result: List<CompactFormats.CompactBlock>) = withContext(SdkDispatchers.IO) {
+    override suspend fun write(result: List<CompactFormats.CompactBlock>) =
         cacheDao.insert(result.map { CompactBlockEntity(it.height.toInt(), it.toByteArray()) })
-    }
 
-    override suspend fun rewindTo(height: Int) = withContext(SdkDispatchers.IO) {
+    override suspend fun rewindTo(height: Int) =
         cacheDao.rewindTo(height)
-    }
 
     override suspend fun close() {
-        withContext(SdkDispatchers.IO) {
+        withContext(SdkDispatchers.DATABASE_IO) {
             cacheDb.close()
         }
     }
+
+    companion object {
+        /**
+         * @param appContext the application context. This is used for creating the database.
+         * @property dbPath the absolute path to the database.
+         */
+        fun new(appContext: Context, dbPath: String): CompactBlockDbStore {
+            val cacheDb = createCompactBlockCacheDb(appContext.applicationContext, dbPath)
+            val cacheDao = cacheDb.compactBlockDao()
+
+            return CompactBlockDbStore(cacheDao, cacheDb)
```

Comment:
See previous comment above. Could someone create an instance of this object that refers to different databases?